rustler_tests/test/codegen_test.exs

```diff
@@ -136,6 +136,22 @@ defmodule RustlerTest.CodegenTest do
              assert_raise(ErlangError, fn -> RustlerTest.unit_enum_echo(:somethingelse) end)
   end
 
+  test "tagged enum transcoder 1" do
+    assert {:foo, %{x: 1, y: 2}} ==
+             RustlerTest.tagged_enum_1_echo({:foo, %{x: 1, y: 2}})
+
+    assert {:bar, "hello"} == RustlerTest.tagged_enum_1_echo({:bar, "hello"})
+    assert {:baz, "world"} == RustlerTest.tagged_enum_1_echo({:baz, "world"})
+    assert :qux == RustlerTest.tagged_enum_1_echo(:qux)
```

Comment:
Please also add tests for failing decode. Each sub-type should be tested for error behaviour, e.g. `{:foo, "not a map"}`, `{:bar %{a: :map}}`, `{:baz, 10}`, `{:qux, :not_even_a_variant}`, `{:foo, :too, :many, :elements}`.

Reply:
What error type would be suited for these cases? `ErlangError` or `ArgumentError`?

Reply:
Error handling should probably be quite similar to what `NifUntaggedEnum` returns if no variant matches